calcculator1.c

```diff
@@ -1,171 +1,196 @@
-
-            switch (c) {
-                case '+':
-                             digit_store[++top] = num2 + num1; 
-                            break;
-                case '-': 
-                            digit_store[++top] = num2 - num1; 
-                            break;
-                case '*': 
-                            digit_store[++top] = num2 * num1; 
-                            break;
-                case '/': 
-                        if (num1 == 0) {
-                            printf("Division by zero error\n");
-                            return 0;
-                        }
-                        digit_store[++top] = num2 / num1; 
-                        break;
-            }
-        }
-    }
-    return digit_store[top];
-}
-
-int main() {
-    char exp[100];
-    char expression[100];
-    printf("Please Enter the Expression: ");
-    gets(expression);
-    printf("%s\n",expression);
-    space_remover(expression,exp);
-    if (!isvalid(exp)) {
-        printf("Error: Invalid expression\n");
-        return 0;
-    }
-
-    char* postfix = infixToPostfix(exp);
-    //printf("Postfix is: %s\n", postfix);//
-
-    int result = postfixEvaluation(postfix);
-    printf("Result is: %d\n", result);
-
-    
-    return 0;
-}
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+
+/*To check character is operator or not*/
+int isOperator(char c) {
+    return (c == '+' || c == '-' || c == '*' || c == '/');
+}
+int isSpace(char c)
+{
+    return ( c==' ');
+}
+
+int isDigit(char c)
+{
+    if(c>='0' && c<='9')
+    {
+        return 1;
+    }
+    else{
+        return 0;
+    }
+}
+int findLen(char expression[])
+{   int length =0;
+    int i=0;
+    while(expression[i]!='\0')
+    {
+        i++;
+        length++;
+    }
+    return length;
+}
+
+/* To give different precedence  to different operator*/
+int precedence(char c) {
+    if (c == '/')
+        return 4;
+    else if (c == '*')
+        return 3;
+    else if (c == '+')
+        return 2;
+    else
+        return 1;
+}
+
+/* If expression contain whitespace then remove it*/
+void spaceRemover(char expression[], char* exp)
+{
+    int len =findLen(expression);
+    int j =0;
+    for(int i =0;i<len;i++)
+    {
+        if(expression[i]!=' ')
+        {
+            exp[j]=expression[i];
+            j++;
+        }
+    }
+    exp[j]='\0';
+}
+
+/* To check expression is valid or not*/
+int isValid(char exp[]) {
+    int length = findLen(exp);
+    if (length == 0) /* expression should not be empty*/
+       { return 0;}
+    if ((isOperator(exp[0]) && exp[0]!='-') || isOperator(exp[length - 1]))  /* first and last character should not operator , except first can be '-'*/
+       { return 0;}
+
+    for (int i = 0; i < length; i++) {
+        char c = exp[i];
+        if (!isDigit(c) && !isOperator(c) && !isSpace(c) ) {  /* check expression contain only digit, operator, space*/
+            return 0;

```

Comment:
do the same here, Inlcude printf statement so that user also aware of what is going on in the program.